extras/nightly-container/provision.yml

```diff
@@ -77,11 +77,37 @@
       args:
         warn: no
 
+    - name: Create override directory for glusterd2.service
+      file:
+        path: /etc/systemd/system/glusterd2.service.d
+        state: directory
+
+    # This makes systemd pass environment variables set for GD2 by kubernetes
+    - name: Enable PassEnvironment override for glusterd2.service
+      ini_file:
+        path: /etc/systemd/system/glusterd2.service.d/override.conf
+        section: Service
+        option: PassEnvironment
+        # PassEnvironment requires that each variable be mentioned individually
+        # More variables will need to be added here as required
+        value: GD2_ETCDENDPOINTS GD2_CLUSTER_ID GD2_RESTAUTH
+
+    - name: Create /etc/sysconfig/glusterd2/
+      file:
+        path: /etc/sysconfig/glusterd2
+        state: directory
+
+    - name: Disable embedded etcd for GD2
+      lineinfile:
+        path: /etc/sysconfig/glusterd2/noembed
```

Comment:
This is being configured as an environment variable here too. But since this is gonna be the default for all launched containers, we're setting this with the image itself. The other options that we pass as pod environment variables change for each pod.